targets/csharp/source/source/PlayFabGameserverInternal.cs

```diff
@@ -0,0 +1,324 @@
+﻿// GSDK Only has the following supported frameworks for now
+#if NETSTANDARD2_0 || NETCOREAPP2_1
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PlayFab
+{
+    class GameserverInternalSdk : IDisposable
+    {
+        private string _overrideConfigFileName;
+        private GameState _state;
+
+        private Task _heartbeatTask;
+        private GameserverConfiguration _configuration;
+        private IGameserverHttpClient _webClient;
+        private bool _heartbeatRunning;
+        private DateTime _cachedScheduleMaintDate;
+        private ManualResetEvent _heartbeatDoneEvent = new ManualResetEvent(false);
+        private ManualResetEvent _signalHeartbeatEvent = new ManualResetEvent(false);
+        private bool _debug;
+
+        public ManualResetEvent TransitionToActiveEvent { get; set; }
+        public IDictionary<string, string> ConfigMap { get; private set; }
+        public ILogger Logger { get; private set; }
+        public GameState State
+        {
+            get { return _state; }
+            set
+            {
+                if (_state != value)
+                {
+                    _state = value;
+                    _signalHeartbeatEvent.Set();
+                }
+            }
+        }
+
+        public IList<GameserverConnectedPlayer> ConnectedPlayers { get; set; }
+
+        public IList<string> InitialPlayers { get; set; }
+
+        // Keep an instance of the callbacks around so they don't get garbage collected
+        public Action ShutdownCallback { get; set; }
+        public Func<bool> HealthCallback { get; set; }
+        public Action<DateTimeOffset> MaintenanceCallback { get; set; }
+
+
+        public GameserverInternalSdk(string configFileName = null)
+        {
+            _overrideConfigFileName = configFileName;
+            ConnectedPlayers = new List<GameserverConnectedPlayer>();
+            InitialPlayers = new List<string>();
+            TransitionToActiveEvent = new ManualResetEvent(false);
+        }
+
+        public Task StartAsync(bool debugLogs = false)
+        {
+            // If we already initialized everything, no need to do it again
+            if (_heartbeatTask != null)
+            {
+                return Task.CompletedTask;
+            }
+
+            _debug = debugLogs;
+
+
+            if (_configuration == null)
+            {
+                _configuration = GetConfiguration();
```

Comment:
gotta use all the new syntactic sugar  _ configuration = _configuration ?? GetConfiguration()

Reply:
here and below

---
In reply to: [255765818](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r255765818) [](ancestors = 255765818)